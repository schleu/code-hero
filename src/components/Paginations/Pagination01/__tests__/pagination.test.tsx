import "@testing-library/jest-dom/vitest";
import { render, fireEvent, screen } from "@testing-library/react";
import { expect } from "vitest";
import { Pagination } from "../index";

describe("<Pagination />", () => {
  it("Pagination Snapshot 1-10", () => {
    const component = render(
      <Pagination actual={1} total={10} getActualPage={() => {}} />
    );
    expect(component).toMatchSnapshot("Default 1-10");
  });

  it("Pagination Snapshot 5-10", () => {
    const component = render(
      <Pagination actual={5} total={10} getActualPage={() => {}} />
    );
    expect(component).toMatchSnapshot("Default  5-10");
  });

  it("Pagination Snapshot 10-10", () => {
    const component = render(
      <Pagination actual={10} total={10} getActualPage={() => {}} />
    );
    expect(component).toMatchSnapshot("Default  10-10");
  });

  it("Should be have pagination itens and fake itens", () => {
    const component = render(
      <Pagination actual={1} total={10} getActualPage={() => {}} />
    );

    const item = component.getAllByTestId("paginationItem");
    const fake = component.getAllByTestId("fakeItem");

    expect(item).toHaveLength(4);
    expect(fake).toHaveLength(4);

    expect(component.getAllByText("1")).toHaveLength(1);
    expect(component.getAllByText("2")).toHaveLength(1);
    expect(() => component.getAllByText("3")).not.toHaveLength(1);
    expect(component.getAllByText("...")).toHaveLength(1);
    expect(component.getAllByText("10")).toHaveLength(1);
  });

  it("Should be able to next page", () => {
    const component = render(
      <Pagination actual={1} total={10} getActualPage={() => {}} />
    );

    const item = component.getAllByTestId("paginationItem");
    expect(item).toHaveLength(4);

    expect(() => component.getAllByText("3")).toThrow();
    expect(() => component.getAllByText("3")).not.toHaveLength(1);

    const nextPage = component.getAllByTestId("nextPage");
    expect(nextPage).toHaveLength(1);

    fireEvent.click(screen.getByTestId("nextPage"));

    expect(component.getAllByText("1")).toHaveLength(1);
    expect(component.getAllByText("2")).toHaveLength(1);
    expect(component.getAllByText("3")).toHaveLength(1);
    expect(component.getAllByText("...")).toHaveLength(1);
    expect(component.getAllByText("10")).toHaveLength(1);
  });

  it("Should be able to previous page", () => {
    const component = render(
      <Pagination actual={3} total={10} getActualPage={() => {}} />
    );

    const item = component.getAllByTestId("paginationItem");
    expect(item).toHaveLength(6);

    expect(component.getAllByText("1")).toHaveLength(1);
    expect(component.getAllByText("2")).toHaveLength(1);
    expect(component.getAllByText("3")).toHaveLength(1);
    expect(component.getAllByText("...")).toHaveLength(1);
    expect(component.getAllByText("10")).toHaveLength(1);

    const previousPage = component.getAllByTestId("previousPage");
    expect(previousPage).toHaveLength(1);

    fireEvent.click(screen.getByTestId("previousPage"));

    expect(component.getAllByText("1")).toHaveLength(1);
    expect(component.getAllByText("2")).toHaveLength(1);
    expect(() => component.getAllByText("3")).not.toHaveLength(1);
    expect(component.getAllByText("...")).toHaveLength(1);
    expect(component.getAllByText("10")).toHaveLength(1);
  });

  it("Should be able to change especific page", () => {
    const component = render(
      <Pagination actual={3} total={10} getActualPage={() => {}} />
    );

    const item = component.getAllByTestId("paginationItem");
    expect(item).toHaveLength(6);

    expect(component.getAllByText("1")).toHaveLength(1);
    expect(component.getAllByText("2")).toHaveLength(1);
    expect(component.getAllByText("3")).toHaveLength(1);
    expect(component.getAllByText("...")).toHaveLength(1);
    expect(component.getAllByText("10")).toHaveLength(1);

    const previousPage = component.getAllByTestId("previousPage");
    expect(previousPage).toHaveLength(1);
    const nextPage = component.getAllByTestId("nextPage");
    expect(nextPage).toHaveLength(1);

    fireEvent.click(screen.getByText("10"));

    expect(() => component.getAllByText("1")).not.toHaveLength(1);
    expect(() => component.getAllByText("2")).not.toHaveLength(1);
    expect(() => component.getAllByText("3")).not.toHaveLength(1);
    expect(component.getAllByText("...")).toHaveLength(1);
    expect(component.getAllByText("9")).toHaveLength(1);
    expect(component.getAllByText("10")).toHaveLength(1);

    expect(() => component.getAllByTestId("nextPage")).toThrow();
    expect(() => component.getAllByTestId("nextPage")).not.toHaveLength(1);
  });
});
