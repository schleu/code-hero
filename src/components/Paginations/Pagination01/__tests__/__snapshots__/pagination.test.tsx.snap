// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Pagination /> > Pagination Snapshot 1-10 > Default 1-10 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="pagination"
      >
        <div
          class="paginationItem fakeItem"
          data-testid="fakeItem"
        />
        <div
          class="paginationItem fakeItem"
          data-testid="fakeItem"
        />
        <div
          class="paginationItem paginationItemSelected paginationNumber "
          data-testid="paginationNumber"
        >
          1
        </div>
        <div
          class="paginationItem  paginationNumber "
          data-testid="paginationNumber"
        >
          2
        </div>
        <div
          class="paginationItem  paginationNumber "
          data-testid="paginationNumber"
        >
          3
        </div>
        <div
          class="paginationItem  paginationNumber "
          data-testid="paginationNumber"
        >
          4
        </div>
        <div
          class="paginationItem  paginationNumber "
          data-testid="paginationNumber"
        >
          5
        </div>
        <div
          class="paginationItem   "
          data-testid="paginationItem"
        >
          <svg
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8.25 4.5l7.5 7.5-7.5 7.5"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </div>
        <div
          class="paginationItem   "
          data-testid="paginationItem"
        >
          <svg
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.25 4.5l7.5 7.5-7.5 7.5m-6-15l7.5 7.5-7.5 7.5"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="pagination"
    >
      <div
        class="paginationItem fakeItem"
        data-testid="fakeItem"
      />
      <div
        class="paginationItem fakeItem"
        data-testid="fakeItem"
      />
      <div
        class="paginationItem paginationItemSelected paginationNumber "
        data-testid="paginationNumber"
      >
        1
      </div>
      <div
        class="paginationItem  paginationNumber "
        data-testid="paginationNumber"
      >
        2
      </div>
      <div
        class="paginationItem  paginationNumber "
        data-testid="paginationNumber"
      >
        3
      </div>
      <div
        class="paginationItem  paginationNumber "
        data-testid="paginationNumber"
      >
        4
      </div>
      <div
        class="paginationItem  paginationNumber "
        data-testid="paginationNumber"
      >
        5
      </div>
      <div
        class="paginationItem   "
        data-testid="paginationItem"
      >
        <svg
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8.25 4.5l7.5 7.5-7.5 7.5"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
      <div
        class="paginationItem   "
        data-testid="paginationItem"
      >
        <svg
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11.25 4.5l7.5 7.5-7.5 7.5m-6-15l7.5 7.5-7.5 7.5"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<Pagination /> > Pagination Snapshot 5-10 > Default  5-10 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="pagination"
      >
        <div
          class="paginationItem   "
          data-testid="paginationItem"
        >
          <svg
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18.75 19.5l-7.5-7.5 7.5-7.5m-6 15L5.25 12l7.5-7.5"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </div>
        <div
          class="paginationItem   "
          data-testid="paginationItem"
        >
          <svg
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15.75 19.5L8.25 12l7.5-7.5"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </div>
        <div
          class="paginationItem  paginationNumber "
          data-testid="paginationNumber"
        >
          1
        </div>
        <div
          class="paginationItem  paginationNumber "
          data-testid="paginationNumber"
        >
          2
        </div>
        <div
          class="paginationItem  paginationNumber "
          data-testid="paginationNumber"
        >
          3
        </div>
        <div
          class="paginationItem  paginationNumber "
          data-testid="paginationNumber"
        >
          4
        </div>
        <div
          class="paginationItem paginationItemSelected paginationNumber "
          data-testid="paginationNumber"
        >
          5
        </div>
        <div
          class="paginationItem   "
          data-testid="paginationItem"
        >
          <svg
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8.25 4.5l7.5 7.5-7.5 7.5"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </div>
        <div
          class="paginationItem   "
          data-testid="paginationItem"
        >
          <svg
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.25 4.5l7.5 7.5-7.5 7.5m-6-15l7.5 7.5-7.5 7.5"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="pagination"
    >
      <div
        class="paginationItem   "
        data-testid="paginationItem"
      >
        <svg
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18.75 19.5l-7.5-7.5 7.5-7.5m-6 15L5.25 12l7.5-7.5"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
      <div
        class="paginationItem   "
        data-testid="paginationItem"
      >
        <svg
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.75 19.5L8.25 12l7.5-7.5"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
      <div
        class="paginationItem  paginationNumber "
        data-testid="paginationNumber"
      >
        1
      </div>
      <div
        class="paginationItem  paginationNumber "
        data-testid="paginationNumber"
      >
        2
      </div>
      <div
        class="paginationItem  paginationNumber "
        data-testid="paginationNumber"
      >
        3
      </div>
      <div
        class="paginationItem  paginationNumber "
        data-testid="paginationNumber"
      >
        4
      </div>
      <div
        class="paginationItem paginationItemSelected paginationNumber "
        data-testid="paginationNumber"
      >
        5
      </div>
      <div
        class="paginationItem   "
        data-testid="paginationItem"
      >
        <svg
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8.25 4.5l7.5 7.5-7.5 7.5"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
      <div
        class="paginationItem   "
        data-testid="paginationItem"
      >
        <svg
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11.25 4.5l7.5 7.5-7.5 7.5m-6-15l7.5 7.5-7.5 7.5"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<Pagination /> > Pagination Snapshot 10-10 > Default  10-10 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="pagination"
      >
        <div
          class="paginationItem   "
          data-testid="paginationItem"
        >
          <svg
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18.75 19.5l-7.5-7.5 7.5-7.5m-6 15L5.25 12l7.5-7.5"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </div>
        <div
          class="paginationItem   "
          data-testid="paginationItem"
        >
          <svg
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15.75 19.5L8.25 12l7.5-7.5"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </div>
        <div
          class="paginationItem  paginationNumber "
          data-testid="paginationNumber"
        >
          6
        </div>
        <div
          class="paginationItem  paginationNumber "
          data-testid="paginationNumber"
        >
          7
        </div>
        <div
          class="paginationItem  paginationNumber "
          data-testid="paginationNumber"
        >
          8
        </div>
        <div
          class="paginationItem  paginationNumber "
          data-testid="paginationNumber"
        >
          9
        </div>
        <div
          class="paginationItem paginationItemSelected paginationNumber "
          data-testid="paginationNumber"
        >
          10
        </div>
        <div
          class="paginationItem fakeItem"
          data-testid="fakeItem"
        />
        <div
          class="paginationItem fakeItem"
          data-testid="fakeItem"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="pagination"
    >
      <div
        class="paginationItem   "
        data-testid="paginationItem"
      >
        <svg
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18.75 19.5l-7.5-7.5 7.5-7.5m-6 15L5.25 12l7.5-7.5"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
      <div
        class="paginationItem   "
        data-testid="paginationItem"
      >
        <svg
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.75 19.5L8.25 12l7.5-7.5"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
      <div
        class="paginationItem  paginationNumber "
        data-testid="paginationNumber"
      >
        6
      </div>
      <div
        class="paginationItem  paginationNumber "
        data-testid="paginationNumber"
      >
        7
      </div>
      <div
        class="paginationItem  paginationNumber "
        data-testid="paginationNumber"
      >
        8
      </div>
      <div
        class="paginationItem  paginationNumber "
        data-testid="paginationNumber"
      >
        9
      </div>
      <div
        class="paginationItem paginationItemSelected paginationNumber "
        data-testid="paginationNumber"
      >
        10
      </div>
      <div
        class="paginationItem fakeItem"
        data-testid="fakeItem"
      />
      <div
        class="paginationItem fakeItem"
        data-testid="fakeItem"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
